packages/block-library/src/gallery/edit.js

```diff
@@ -473,8 +473,8 @@ export default compose( [
 	withDispatch( ( dispatch, { clientId }, { select } ) => {
 		return {
 			wasBlockJustInserted() {
-				const { clearLastBlockInserted } = dispatch( 'core/editor' );
-				const { wasBlockJustInserted } = select( 'core/editor' );
+				const { clearLastBlockInserted } = dispatch( blockEditorStore );
+				const { wasBlockJustInserted } = select( blockEditorStore );
 
 				const result = wasBlockJustInserted( clientId );
 
```

Comment:
@fluiddot This is the block of code that is causing the issue. What I am trying to accomplish here is, once I get the results of `wasBlockJustInserted` I need to clear the value that marks the current Gallery as the last block inserted so undo/redo and other state changes do not trigger the opening of the sheet again. 

How do you think I can refactor this? I know that the codebase has hooks such as `useSelect` and `useDispatch` and I was wondering if memorization might be able to help here but I wasn't sure. Let me know what you think.

Reply:
Yeah, I think the best option here would be to apply a refactor. I posted a [comment](https://github.com/WordPress/gutenberg/pull/30123#issuecomment-805024819) related to this.